### PR TITLE
[12.0] mig stock_picking_filter_lot

### PR DESCRIPTION
**Pull request by IsabelEsparza**

### CODE REVIEW

**Review by HaraldPanten — APPROVED** [review]
Functional review on runbot 👍

File: stock_picking_filter_lot/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock picking filter lot",
+    "summary": "In picking out lots' selection, "
+               "filter lots based on their location",
+    "version": "12.0.1.0.0",
+    "category": "Warehouse",
+    "website": "https://github.com/OCA/stock-logistics-workflow/tree/"
```
Comment (tbaden) [inline stock_picking_filter_lot/__manifest__.py]: website should be: https://github.com/OCA/stock-logistics-workflow

**Review by tbaden — CHANGES_REQUESTED** [review]
thanks for the work 👍 
I did a code review: LGTM, just a minor change.

File: stock_picking_filter_lot/__manifest__.py (hunk 1 above)
Comment (IsabelEsparza) [reply stock_picking_filter_lot/__manifest__.py]: I just changed it.

**Review by alan196 — APPROVED** [review]
Functional Review! 👍

File: stock_picking_filter_lot/views/stock_picking_view.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
         <field name="model">stock.picking</field>
         <field name="inherit_id" ref="stock.view_picking_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='move_line_ids']/tree/field[@name='lot_id']" position="attributes">
+            <xpath expr="//field[@name='move_line_ids_without_package']/tree/field[@name='lot_id']" position="attributes">
```
Comment (LoisRForgeFlow) [inline stock_picking_filter_lot/views/stock_picking_view.xml:12]: why this change?

**Review by LoisRForgeFlow — COMMENTED** [review]
Functioanl test :+1: 

Comment inline:

File: stock_picking_filter_lot/views/stock_picking_view.xml (hunk 2 above)
Comment (tbaden) [reply stock_picking_filter_lot/views/stock_picking_view.xml:12]: @lreficent move_line_ids no longer exists this way in the record. It got changed to move_line_ids_without_package

**Review by tbaden — APPROVED** [review]

File: stock_picking_filter_lot/views/stock_picking_view.xml (hunk 2 above)
Comment (LoisRForgeFlow) [reply stock_picking_filter_lot/views/stock_picking_view.xml:12]: Oh, I see, you are right. Thanks.

**Comment by damendieta** [comment]
Hi, I found a (kind of) bug in this PR.

As this module computes the filter based on quant location, there is a problem when you have a quant that has been transfered and it's quantity is 0.

I have tried a modification with this changes:

```python
    @api.depends('quant_ids', 'quant_ids.location_id', 'quant_ids.quantity')
    def _compute_location_ids(self):
        for lot in self:
            lot.location_ids = lot.mapped('quant_ids') \
                .filtered(lambda q: q.quantity > 0).mapped('location_id') \
                .filtered(lambda l: l.usage == 'internal')
```

It gives a more real filter, even excludes not internal locations (to avoid vendor and customer locations).

I don't know if a new PR is worth it, but I leave this suggestion here if someone needs it.

**Comment by LoisRForgeFlow** [comment]
@damendieta Thanks for noticing, I think it is worth, so please open a PR with the proposed fix.
